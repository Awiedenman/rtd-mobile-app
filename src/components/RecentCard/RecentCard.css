.recent-button {
  height: 60px;
  margin-top: 5px;
  width: 95%;
  border-radius: 5px;
  box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.5);
  background: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.address-container {
  display: flex;
}

.from-to {
  font-family: 'Open Sans', sans-serif;
  font-weight: 600;
  font-size: 15px;
  margin: 5px;
}

.address {
  font-family: 'Open Sans', sans-serif;
  font-size: 15px;
  margin: 5px;
}
