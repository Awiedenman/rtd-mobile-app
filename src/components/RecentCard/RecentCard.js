import React from 'react';
import PropTypes from 'prop-types';

import './RecentCard.css';

const RecentCard = (props) => {
  console.log(props)
  const {
    recentData,
    searchRecent
  } = props;

  const handleClick = () => {
    const {
      start_address,
      end_address,
    } = recentData;
    const recentInfo = {
      start_address,
      end_address
    }
    searchRecent(recentInfo);
  }
  
  return (
    <button
      className='recent-button'
      onClick={handleClick}
    >
      <div className='address-container'>
        <p className='from-to'>From:</p>
        <p className='address'>
          {recentData.start_address.split(',')[0]}
        </p>
      </div>
      <div className='address-container'>
        <p className='from-to'>To:</p>
        <p className='address'>
          {recentData.end_address.split(',')[0]}
        </p>
      </div>
    </button>
  );
};

export default RecentCard;

RecentCard.propTypes = {
  recentData: PropTypes.array,
  searchRecent: PropTypes.func
};