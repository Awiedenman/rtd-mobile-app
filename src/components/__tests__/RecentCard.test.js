import React from 'react';
import { shallow } from 'enzyme';
import RecentCard from '../RecentCard/RecentCard';


describe('RecentCard', () => {
  it('should match snapshot', () => {
    const mockProps = {
      recentData: {
        end_address: "Union Station, Denver, CO 80202, USA",
        favorite: false,
        id: 199,
        start_address: "3069 S Ogden St, Englewood, CO 80113, USA"
      },
      searchRecent: jest.fn()
    };
    const mockRecentData = {
      end_address: "Union Station, Denver, CO 80202, USA",
      favorite: false,
      id: 199,
      start_address: "3069 S Ogden St, Englewood, CO 80113, USA"
    }

    const wrapper = shallow(
      <RecentCard
        props={mockProps}
        recentData={mockRecentData}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});
