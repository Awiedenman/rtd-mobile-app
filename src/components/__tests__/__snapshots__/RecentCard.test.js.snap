// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecentCard
    props={
        Object {
            "recentData": Object {
              "end_address": "Union Station, Denver, CO 80202, USA",
              "favorite": false,
              "id": 199,
              "start_address": "3069 S Ogden St, Englewood, CO 80113, USA",
            },
            "searchRecent": [Function],
          }
    }
    recentData={
        Object {
            "end_address": "Union Station, Denver, CO 80202, USA",
            "favorite": false,
            "id": 199,
            "start_address": "3069 S Ogden St, Englewood, CO 80113, USA",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="address-container"
>
          <p
                    className="from-to"
          >
                    From:
          </p>
          <p
                    className="address"
          >
                    3069 S Ogden St
          </p>
</div>,
        <div
          className="address-container"
>
          <p
                    className="from-to"
          >
                    To:
          </p>
          <p
                    className="address"
          >
                    Union Station
          </p>
</div>,
      ],
      "className": "recent-button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="from-to"
>
              From:
</p>,
            <p
              className="address"
>
              3069 S Ogden St
</p>,
          ],
          "className": "address-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "From:",
              "className": "from-to",
            },
            "ref": null,
            "rendered": "From:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3069 S Ogden St",
              "className": "address",
            },
            "ref": null,
            "rendered": "3069 S Ogden St",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="from-to"
>
              To:
</p>,
            <p
              className="address"
>
              Union Station
</p>,
          ],
          "className": "address-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "To:",
              "className": "from-to",
            },
            "ref": null,
            "rendered": "To:",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Union Station",
              "className": "address",
            },
            "ref": null,
            "rendered": "Union Station",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="address-container"
>
            <p
                        className="from-to"
            >
                        From:
            </p>
            <p
                        className="address"
            >
                        3069 S Ogden St
            </p>
</div>,
          <div
            className="address-container"
>
            <p
                        className="from-to"
            >
                        To:
            </p>
            <p
                        className="address"
            >
                        Union Station
            </p>
</div>,
        ],
        "className": "recent-button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="from-to"
>
                From:
</p>,
              <p
                className="address"
>
                3069 S Ogden St
</p>,
            ],
            "className": "address-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From:",
                "className": "from-to",
              },
              "ref": null,
              "rendered": "From:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "3069 S Ogden St",
                "className": "address",
              },
              "ref": null,
              "rendered": "3069 S Ogden St",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="from-to"
>
                To:
</p>,
              <p
                className="address"
>
                Union Station
</p>,
            ],
            "className": "address-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "To:",
                "className": "from-to",
              },
              "ref": null,
              "rendered": "To:",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Union Station",
                "className": "address",
              },
              "ref": null,
              "rendered": "Union Station",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
